packages/sdk/src/browser-wallets/leather/index.ts

```diff
@@ -47,7 +47,14 @@ async function getAddresses(
 
   // Hacky validation: there's no parameter to specify the network value when getting the address.
   // TODO: Remove this if the wallet already supports that parameter.
-  if (getNetworkByAddress(addresses[0].address) !== network) {
+  const derivedNetwork = getNetworkByAddress(addresses[0].address);
+  if (network !== "signet" && derivedNetwork !== network) {
```

Comment:
i think these 2 `if` blocks can be combined but the 2nd condition should be labelled that derived network will return testnet since regtest and testnet are equivalent